soc/espressif/esp32/memory.h

```diff
@@ -5,22 +5,22 @@
 #pragma once
 
 /* SRAM0 (192kB)  instruction cache+memory */
-#define SRAM0_IRAM_START    0x40070000
+#define SRAM0_IRAM_START    DT_REG_ADDR(DT_NODELABEL(sram0))
 #define SRAM0_CACHE_SIZE    0x10000
-#define SRAM0_SIZE          0x30000
+#define SRAM0_SIZE          DT_REG_SIZE(DT_NODELABEL(sram0))
 
 /* SRAM1 (128kB) instruction/data memory */
-#define SRAM1_IRAM_START    0x400a0000
-#define SRAM1_DRAM_START    0x3ffe0000
-#define SRAM1_SIZE          0x20000
+#define SRAM1_IRAM_START    (SRAM0_IRAM_START + SRAM0_SIZE)
+#define SRAM1_DRAM_START    DT_REG_ADDR(DT_NODELABEL(sram1))
+#define SRAM1_SIZE          DT_REG_SIZE(DT_NODELABEL(sram1))
 #define SRAM1_DRAM_END      (SRAM1_DRAM_START + SRAM1_SIZE)
 #define SRAM1_DRAM_PROAPP_PRIV_SIZE 0x8000
 #define SRAM1_DRAM_USER_START 0x3ffe8000
```

Comment:
nit: this could be defined as DT_REG_ADDR+PROAPP_PRIV_SIZE so it is not hardcoded.